app/domain/errors.rb

```diff
@@ -0,0 +1,213 @@
+        msg: "Invalid credentials",
+        code: "CONJ00002E"
+      )
+
+      InvalidOrigin = ::Util::ErrorClassWithCode.new(
+        msg: "Invalid origin",
+        code: "CONJ00003E"
+      )
+
+      module Security
+
+        NotWhitelisted = ::Util::ErrorClassWithCode.new(
+          msg: "'{0}' not whitelisted in CONJUR_AUTHENTICATORS",
+          code: "CONJ00004E"
+        )
+
+        ServiceNotDefined = ::Util::ErrorClassWithCode.new(
+          msg: "Webservice '{0}' is not defined in the Conjur policy",
+          code: "CONJ00005E"
+        )
+
+        UserNotAuthorizedInConjur = ::Util::ErrorClassWithCode.new(
+          msg: "User '{0}' is not authorized in the Conjur policy",
+          code: "CONJ00006E"
+        )
+
+        UserNotDefinedInConjur = ::Util::ErrorClassWithCode.new(
+          msg: "User '{0}' is not defined in Conjur",
+          code: "CONJ00007E"
+        )
+
+        AccountNotDefined = ::Util::ErrorClassWithCode.new(
+          msg: "account '{0}' is not defined in Conjur",
+          code: "CONJ00008E"
+        )
+
+      end
+
+      module RequestBody
+
+        MissingRequestParam = ::Util::ErrorClassWithCode.new(
+          msg: "field '{0}' is missing or empty in request body",
+          code: "CONJ00009E"
+        )
+
+      end
+
+      module AuthnOidc
+
+        ProviderDiscoveryTimeout = ::Util::ErrorClassWithCode.new(
+          msg: "OIDC provider discovery failed with timeout error (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00010E"
+        )
+
+        ProviderDiscoveryFailed = ::Util::ErrorClassWithCode.new(
+          msg: "OIDC provider discovery failed (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00011E"
+        )
+
+        ProviderFetchCertificateFailed = ::Util::ErrorClassWithCode.new(
+          msg: "Failed to fetch certificate from OIDC provider (provider_uri='{0}'). Reason: '{1}'",
+          code: "CONJ00012E"
+        )
+
+        IdTokenFieldNotFoundOrEmpty = ::Util::ErrorClassWithCode.new(
+          msg: "Field '{0}' not found or empty in ID Token",
+          code: "CONJ00013E"
+        )
+
+        IdTokenInvalidFormat = ::Util::ErrorClassWithCode.new(
+          msg: "Invalid ID Token Format (3rdPartyError ='{0}')",
+          code: "CONJ00014E"
+        )
+
+        IdTokenVerifyFailed = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token verification failed (3rdPartyError ='{0}')",
+          code: "CONJ00015E"
+        )
+
+        IdTokenExpired = ::Util::ErrorClassWithCode.new(
+          msg: "ID Token Expired",
+          code: "CONJ00016E"
+        )
+
+        AdminAuthenticationDenied = ::Util::ErrorClassWithCode.new(
+          msg: "admin user is not allowed to authenticate with OIDC",
+          code: "CONJ00017E"
+        )
+
+      end
+
+      module AuthnIam
+
+        InvalidAWSHeaders = ::Util::ErrorClassWithCode.new(
+          msg: "'Invalid or Expired AWS Headers: {0}",
```

Comment:
Invalid or expired AWS headers - sentence caps (only first word and proper nouns/acronyms are capitalized)